### DISCUSSION TITLE
BUG: Switching pages calls canvas styles and scripts again and again

### DISCUSSION THREAD

**Discussion by chiragkataria22** [GrapesJS/grapesjs#3468] [Ideas]
Hi

I am integrating page manager into my project. But the issue I am facing is that whenever I switch between pages it loads CSS and script files given in canvas object at the time of init, again and again.

It makes the switching between pages slow.

Implementation:

grapesjs.init({
    pageManager: {
        pages: [
          {
            id: 'Page-1',
            styles: ``, // or a JSON of styles
            component: '', // or a JSON of components
            name: 'Page 1',
          },
        ],
      },

      canvas: {
        styles: [
          '../assets/bootstrap/bootstrap.min.css',
          '../assets/platforma/platforma.min.css',
          '../assets/editor.css',
        ],
        scripts: [
          '../assets/bootstrap/jquery-3.5.1.slim.min.js',
          '../assets/bootstrap/bootstrap.bundle.min.js',
          '../assets/platforma/platforma.bundle.min.js',
        ],
      },
})

after that, I am adding pages manually on click of a button

**Comment by chiragkataria22** [comment]
@artf Can you please help?

**Comment by artf** [comment]
Unfortunately, switching pages creates a new iframe so, the behavior is correct.
I preferred to remove previous iframes in order to keep the memory usage as low as possible. One solution might be to find a way to reuse previous iframes but that might cause context issues with external JS... so I'm not sure how to properly prevent this behavior.
Anyway, to be honest I'd not expect switching so slow after the first load as external files should be cached by the browser... are you able to create a screencast of your experience?
